src/vos/evtree.c

```diff
@@ -1823,6 +1843,70 @@ evt_select_node(struct evt_context *tcx, const struct evt_rect *rect,
 	return rc < 0 ? nd1 : nd2;
 }
 
+/** Expand the dynamic tree root node if it is currently full and not
+ *  already at full size
+ */
+static inline int
+evt_node_extend(struct evt_context *tcx, struct evt_trace *trace)
+{
+	struct evt_node *nd_cur;
+	int              rc;
+	uint8_t          old_order;
+	void            *new_node;
+	int              old_size;
+	umem_off_t       new_off;
+
+	if (tcx->tc_order == tcx->tc_max_order)
+		return 0;
+
+	nd_cur = evt_off2node(tcx, trace->tr_node);
+	D_ASSERT(evt_node_is_leaf(tcx, nd_cur));
+	D_ASSERT(tcx->tc_depth == 1);
+	D_ASSERT(tcx->tc_feats & EVT_FEAT_DYNAMIC_ROOT);
```

Comment:
How can this be guaranteed?

Reply:
I think you may be confusing root->tr_max_order which would be 0 with tcx->tc_max_order which will always be the user-specified tree order.   As such, the condition above this should guarantee that this bit is set if we get here.